### DISCUSSION TITLE
How to add JaVers audit to an existing database?

### DISCUSSION THREAD

**Discussion by dmitry-weirdo** [javers/javers#1116] [Q&A]
Good afternoon!

I am actually considering switching our PostgreSQL database audit to JaVers from Hibernate Envers.
It is just a PostgreSQL database, no additional NoSQL database can be added.

Yes, I've read https://javers.org/blog/2017/12/javers-vs-envers-comparision.html.

There are two well-known not implemented but highly demanded features in Envers:

**1)** Get changes between 2 revisions — not natively implemented by Envers, therefore I used JaVers Diff to get changes between my Entity objects.
👍 It looks that JaVers offers this functionality out-of-the-box.

**2)** Insert the initial audit data to the current database, so that the existing records/entities can be fully audited (changes in any properties).

In Envers, I had to do it manually. It is error-prone but pretty straightforward: 
- for each root entity, under one revision for each root entity:
-- insert revisions values into `revinfo` table.
-- Copy the values from main tables to `_aud` tables, for all entities/tables from one entity tree.
-- For one-to-many relations, insert `entity1_entity2_aud` tables,

It is a big amount of work, but just copying the SQL columns is simple.

❓ ❓ **The question is:** Does JaVers audit support the initialization from the current database state out-of-the-box?
Because I'm afraid that the manual insertion of the complex JSON structures used by JaVers would me almost impossible.

**Comment by bartoszwalacik** [comment]
There is no migration tool. All you can do with Javers is exposed via the `Javers.java` interface.
Did you tried simply to bulk commit your objects using `javers.commit()`?
